tests/logger/test_handlers.py

```diff
@@ -0,0 +1,204 @@
+        handler.connect_socket()
+        call_args = socket_mocks.socket_initializer.call_args[0]
+        assert call_args[0] == socket
+        assert call_args[2] == SOCK_STREAM
+        assert call_args[3] == IPPROTO_TCP
+        assert socket_mocks.mock_socket.connect.call_count == 1
+
+    @tls_and_tcp_test
+    def test_connect_socket_when_tcp_or_tls_sets_timeout_for_connection_and_resets(
+        self, socket_mocks, protocol
+    ):
+        handler = NoPrioritySysLogHandler(_TEST_HOST, _TEST_PORT, protocol, None)
+        handler.connect_socket()
+        call_args = socket_mocks.mock_socket.settimeout.call_args_list
+        assert len(call_args) == 2
+        assert call_args[0][0][0] == 10
+        assert call_args[1][0][0] is None
+
+    def test_connect_socket_when_tls_and_given_certs_wraps_socket_with_certs(
+        self, socket_mocks
+    ):
+        handler = NoPrioritySysLogHandler(
+            _TEST_HOST, _TEST_PORT, ServerProtocol.TLS_TCP, _TEST_CERTS
+        )
+        handler.connect_socket()
+        socket_mocks.ssl_wrap_socket_mock.assert_called_once_with(
+            socket_mocks.mock_socket, ca_certs=_TEST_CERTS, cert_reqs=ssl.CERT_REQUIRED,
+        )
+
+    def test_connect_socket_when_tls_and_not_given_certs_wraps_socket_with_cert_none(
+        self, socket_mocks
+    ):
+        handler = NoPrioritySysLogHandler(
+            _TEST_HOST, _TEST_PORT, ServerProtocol.TLS_TCP, None
+        )
+        handler.connect_socket()
+        socket_mocks.ssl_wrap_socket_mock.assert_called_once_with(
+            socket_mocks.mock_socket, ca_certs=None, cert_reqs=ssl.CERT_NONE
+        )
+
+    @tls_and_tcp_test
+    def test_emit_when_tcp_calls_socket_sendall_with_expected_message(
+        self, mock_file_event_log_record, protocol
+    ):
+        handler = NoPrioritySysLogHandler(_TEST_HOST, _TEST_PORT, protocol, None)
+        handler.connect_socket()
+        formatter = FileEventDictToRawJSONFormatter()
+        handler.setFormatter(formatter)
+        handler.emit(mock_file_event_log_record)
+        expected_message = (formatter.format(mock_file_event_log_record) + "\n").encode(
+            "utf-8"
+        )
+        handler.socket.sendall.assert_called_once_with(expected_message)
+
+    def test_emit_when_udp_calls_socket_sendto_with_expected_message_and_address(
+        self, mock_file_event_log_record
+    ):
+        handler = NoPrioritySysLogHandler(
+            _TEST_HOST, _TEST_PORT, ServerProtocol.UDP, None
+        )
+        handler.connect_socket()
+        formatter = FileEventDictToRawJSONFormatter()
+        handler.setFormatter(formatter)
+        handler.emit(mock_file_event_log_record)
+        expected_message = (formatter.format(mock_file_event_log_record) + "\n").encode(
+            "utf-8"
+        )
+        handler.socket.sendto.assert_called_once_with(
+            expected_message, (_TEST_HOST, _TEST_PORT)
+        )
+
+    def test_close_when_using_tls_unwraps_socket(self):
```

Comment:
Added these `close` tests and also verified that unwrapping the socket does indeed make the server happy